lib/datadog/ci/ext/environment.rb

```diff
@@ -324,6 +327,22 @@ def extract_bitrise(env)
           }
         end
 
+        def extract_user_defined_git(env)
+          {
+            Datadog::Ext::Git::TAG_REPOSITORY_URL => env['DD_GIT_REPOSITORY_URL'],
+            Datadog::Ext::Git::TAG_COMMIT_SHA => env['DD_GIT_COMMIT_SHA'],
+            Datadog::Ext::Git::TAG_BRANCH => env['DD_GIT_BRANCH'],
+            Datadog::Ext::Git::TAG_TAG => env['DD_GIT_TAG'],
+            Datadog::Ext::Git::TAG_COMMIT_MESSAGE => env['DD_GIT_COMMIT_MESSAGE'],
+            Datadog::Ext::Git::TAG_COMMIT_AUTHOR_NAME => env['DD_GIT_COMMIT_AUTHOR_NAME'],
+            Datadog::Ext::Git::TAG_COMMIT_AUTHOR_EMAIL => env['DD_GIT_COMMIT_AUTHOR_EMAIL'],
+            Datadog::Ext::Git::TAG_COMMIT_AUTHOR_DATE => env['DD_GIT_COMMIT_AUTHOR_DATE'],
+            Datadog::Ext::Git::TAG_COMMIT_COMMITTER_NAME => env['DD_GIT_COMMIT_COMMITTER_NAME'],
+            Datadog::Ext::Git::TAG_COMMIT_COMMITTER_EMAIL => env['DD_GIT_COMMIT_COMMITTER_EMAIL'],
+            Datadog::Ext::Git::TAG_COMMIT_COMMITTER_DATE => env['DD_GIT_COMMIT_COMMITTER_DATE']
+          }.reject { |_, v| v.nil? || v.empty? }
```

Comment:
I would recommend using `strip` on the values first, as sometimes users will introduce whitespace characters that will make the string "not empty", e.g.:

```
2.5.1 :002 > ' '.empty?
 => false
 ```